### PR TITLE
fix(ui): rework artifact reference validation to align with backend rules. 

### PR DESCRIPTION
**Pull request by vjkumar2756**
Close - #9595 
## Summary
This PR reworks artifact reference validation in the UI (`ReferencesFormGroup` and modal components) to enforce strict validation rules on `groupId`, `artifactId`, `version`, and reference `name` fields in accordance with backend validation criteria (`ArtifactIdValidator.java` and `VersionId.java`).

## Changes
- **Validation Utilities (`ui/ui-app/src/utils/validation.utils.ts`)**:
  - Added `checkVersionValid` matching backend `VersionId` regex (`^[a-zA-Z0-9._\-+]{1,256}$`).
  - Added length limit check (max 512 characters) in `checkIdValid`.
  - Added `validateVersionField` for Form controls.

- **References Form Component (`ReferencesFormGroup.tsx`)**:
  - Replaced basic non-empty check with `validateRefIdField` and `validateRefVersionField`.
  - Updated `isReferencesValid` to validate format rules for all reference fields (`groupId`, `artifactId`, `version`, `name`).

- **Modals (`CreateArtifactModal`, `CreateDraftModal`, `NewDraftFromModal`, `CreateAgentModal`)**:
  - Standardized reference form field state and validation handling.

- **Unit Tests (`validation.utils.test.ts`)**:
  - Added comprehensive test coverage for `checkIdValid`, `checkVersionValid`, `validateField`, and `validateVersionField`.

## How to Test
1. Open the **Create Artifact** or **Create Draft** modal in the UI.
2. Add an Artifact Reference.
3. Verify that invalid characters (e.g. `%`, non-ASCII, spaces in version, or strings exceeding max lengths) trigger visual validation errors and prevent form submission.
4. Verify unit tests pass:
   ```bash
   npm --prefix ui/ui-app test

### CODE REVIEW

File: ui/ui-app/src/app/components/modals/ReferencesFormGroup.tsx [hunk 1]
```diff
@@ -11,22 +11,39 @@ export type ArtifactReferenceFormItem = {
     name: string;
 };
 
-type ValidType = "default" | "success" | "error";
+import { checkIdValid, checkVersionValid, ValidType } from "@utils/validation.utils.ts";
 
-const validateRefField = (value: string): ValidType => {
-    if (value === "") {
+export type ArtifactReferenceFormItem = {
```
Comment (paoloantinori) [inline ui/ui-app/src/app/components/modals/ReferencesFormGroup.tsx]: This second `export type ArtifactReferenceFormItem` duplicates the declaration at the top of the file (the hunk context at lines 11-12 is the tail of the original, which is still there), so TypeScript fails with TS2300 duplicate identifier and the UI build breaks. Note that `npm test` alone will not catch this: vitest transpiles through esbuild and does not type-check, so the suite can stay green while `tsc` fails. Please delete this duplicate (keep the original above) and move the `@utils/validation.utils.ts` import up with the other imports at the top of the file.

File: ui/ui-app/src/utils/validation.utils.ts [hunk 2]
```diff
@@ -15,6 +22,18 @@ export const checkIdValid = (id: string | undefined | null): boolean => {
...
 
+/**
+ * Validates version string against backend rules (VersionId.java):
+ * Regex: [a-zA-Z0-9._\-+]{1,256}
+ */
+export const checkVersionValid = (version: string | undefined | null): boolean => {
+    if (!version) {
+        return true;
+    }
+    const VERSION_REGEX = /^[a-zA-Z0-9._\-+]{1,256}$/;
```
Comment (paoloantinori) [inline ui/ui-app/src/utils/validation.utils.ts:44]: In JavaScript, `$` also matches just before a trailing newline, so `checkVersionValid("1.0.0\n")` returns `true` while the same pattern applied with `matches()` on the server rejects it. A version pasted with a trailing newline clears the UI validation and the request then fails server-side with a 400. Anchor without the loophole, for example `/^[a-zA-Z0-9._\-+]{1,256}(?![\s\S])/`, or explicitly reject `\r`/`\n`, and add a pasted-trailing-newline case to `validation.utils.test.ts`.

File: ui/ui-app/src/app/components/modals/ReferencesFormGroup.tsx [hunk 3]
```diff
@@ -11,22 +11,39 @@ export type ArtifactReferenceFormItem = {
...
     }
     return "success";
 };
 
 /**
- * Returns true if all reference rows have all four fields populated.
+ * Returns true if all reference rows have valid populated fields.
  * Returns true if there are no references (empty list is valid).
  */
 export const isReferencesValid = (items: ArtifactReferenceFormItem[]): boolean => {
```
Comment (paoloantinori) [inline ui/ui-app/src/app/components/modals/ReferencesFormGroup.tsx]: `isReferencesValid` is the core behavior change of this PR, it is exported and pure, and nothing covers it. Please add tests, ideally by moving it (with `validateRefIdField`/`validateRefVersionField`) into `validation.utils.ts` where the new test file already lives. Cases to assert: an empty list returns `true`; a row with `%` in `name`, a non-ASCII `groupId`, a 513-character `artifactId`, or `version` of `"1 0"` or `"1:0"` returns `false`; a fully valid row returns `true`.

File: ui/ui-app/src/app/components/modals/CreateArtifactModal.tsx [hunk 4]
```diff
@@ -568,9 +543,16 @@ export const CreateArtifactModal: FunctionComponent<CreateArtifactModalProps> =
...
                                 placeholder="1.0.0 (optional) will be generated if left blank"
                                 onChange={(_evt, value) => setVersionNumber(value)}
-                                // validated={groupValidated()}
+                                validated={validities.versionNumber}
                             />
-                            <If condition={autoDetectedVersion !== undefined && data.firstVersion?.version === autoDetectedVersion}>
+                            <If condition={!isVersionNumberValid}>
+                                <FormHelperText>
+                                    <HelperText>
+                                        <HelperTextItem variant="error" icon={ <ExclamationCircleIcon /> }>Only alphanumeric characters, dots, underscores, hyphens, and plus signs are allowed (max 256 characters)</HelperTextItem>
```
Comment (paoloantinori) [inline ui/ui-app/src/app/components/modals/CreateArtifactModal.tsx:549]: This error helper block and its message are now copy-pasted in four places (`CreateArtifactModal.tsx:551`, `CreateDraftModal.tsx:446`, `NewDraftFromModal.tsx:212`, `CreateAgentModal.tsx:217`) and they have already drifted: this copy has no trailing period, `CreateAgentModal` adds one and renders it through a different markup variant. Please extract a shared message constant (or a small shared helper component) next to `validateVersionField` in `validation.utils.ts` and use it from all four modals.

File: ui/ui-app/src/app/pages/agents/components/modals/CreateAgentModal.tsx [hunk 5]
```diff
@@ -92,7 +93,7 @@ export const CreateAgentModal: FunctionComponent<CreateAgentModalProps> = (props
     };
 
     const isCoordinatesStepValid = (): boolean => {
-        return checkIdValid(groupId) && checkIdValid(artifactId);
+        return checkIdValid(groupId) && checkIdValid(artifactId) && checkVersionValid(version);
```
Comment (paoloantinori) [inline ui/ui-app/src/app/pages/agents/components/modals/CreateAgentModal.tsx:103]: `checkVersionValid` returns `true` for an empty string, so this gate does not actually require a version. If the version is mandatory in this modal (the other modals label it optional, this one does not), the empty case passes here and surfaces as a server-side error instead of inline validation; please confirm the intent and add an explicit non-empty check if it is required. Separately, `validities.version` is only set in `onChange`, so a programmatic reset of `version` leaves stale error state, while the other three modals recompute validity from the value in an effect.

**Review by paoloantinori — COMMENTED** [review]
One merge blocker: `ui/ui-app/src/app/components/modals/ReferencesFormGroup.tsx` now declares `export type ArtifactReferenceFormItem` twice (the original at the top of the file is still present, the hunk context at lines 11-12 shows its tail), which fails the TypeScript build with TS2300 duplicate identifier, and `npm --prefix ui/ui-app test` will not catch it because vitest strips types without type-checking. Beyond that, consolidating the per-modal validators into `validation.utils.ts` and matching the backend rules is the right call and the new tests are a good start; the remaining findings are major but not structural.

- `ui/ui-app/src/utils/validation.utils.ts:33`: the `$` anchor in JavaScript also matches just before a trailing newline, so `checkVersionValid("1.0.0\n")` returns `true` while the server-side pattern check on the same regex rejects it. A version pasted with a trailing newline clears the UI gate and the create/update call then fails with a 400.
- `ui/ui-app/src/app/components/modals/ReferencesFormGroup.tsx:41`: `isReferencesValid` carries the main behavior change of this PR, is exported and pure, and has no test coverage.
- The version error helper text is now copy-pasted into four modals (`CreateArtifactModal.tsx:551`, `CreateDraftModal.tsx:446`, `NewDraftFromModal.tsx:212`, `CreateAgentModal.tsx:217`) and has already drifted: only `CreateAgentModal` ends the message with a period and it uses a different markup variant.
- `ui/ui-app/src/app/pages/agents/components/modals/CreateAgentModal.tsx:96`: `checkVersionValid("")` returns `true`, so `isCoordinatesStepValid` never actually requires a version. If the version is mandatory in this modal (the other three mark it optional, this one does not), the empty case slips through and surfaces as a server-side error. Also, `validities.version` is only updated in `onChange`, so a programmatic reset of `version` leaves stale validation state, unlike the other modals which recompute from the value in an effect.
- Please confirm the backend really applies `ArtifactIdValidator` to reference `name` (`ReferencesFormGroup.tsx:43`): for protobuf, reference names are import paths, and if the server is looser than `checkIdValid` the UI now rejects references the API would accept.
- Smaller checks while in here: `checkIdValid` looks unused after the refactor in `CreateDraftModal.tsx:47` and `NewDraftFromModal.tsx:23` (drop those imports if so), the `export type { ValidType }` re-export from `CreateArtifactModal.tsx:44` keeps the old import path alive (point callers at `@utils/validation.utils.ts` instead), and confirm `versionNumber` is declared on the `Validities` type since the diff does not show it being added.

**Comment by vjkumar2756** [comment]
@paoloantinori  , thanks for the detailed code review! I have addressed all of your feedback items:

1. **Fixed TS2300 Duplicate Type Blocker**: Removed the duplicate `export type ArtifactReferenceFormItem` in `ReferencesFormGroup.tsx` and moved imports cleanly to the top of the file.

2. **Hardened Version Regex against Trailing Newlines**: Updated `checkVersionValid` in `validation.utils.ts` to explicitly reject newlines (`\n`, `\r`) so pasted trailing newlines won't bypass UI validation.
3. **Comprehensive Unit Tests for `isReferencesValid`**: Centralized reference validation helpers and added test assertions for empty lists, valid rows, invalid character `%`, non-ASCII group IDs, 513-character artifact IDs, and version string edge cases (`"1 0"`, `"1.0.0\n"`).
4. **Normalized Version Helper Text & Imports**: Standardized the error helper text across all 4 modals (`CreateArtifactModal`, `CreateDraftModal`, `NewDraftFromModal`, `CreateAgentModal`) and removed unused imports (`checkIdValid` / `ValidType`).
5. **Modal State & Validation Sync**: Updated `CreateAgentModal` to automatically recompute field validity states when inputs change or reset.

I have  verified all changes locally (`140 / 140` Vitest unit tests passing and TypeScript compilation clean). Could you please take another look when you have a chance? Thanks!

**Comment by paoloantinori** [comment]
Re-read at bc22d2d — both items are in and verified:

- The duplicate `export type ArtifactReferenceFormItem` is gone from `ReferencesFormGroup.tsx`: the type now lives in `validation.utils.ts` and is re-exported from the form group, so the TS2300 blocker is resolved.
- `checkVersionValid` now explicitly rejects newline characters before the regex test, with a comment explaining why (JS `$` matches before a trailing newline, so the anchored regex alone would let a pasted trailing newline through). Good catch on that subtlety.

Nothing further from me — thanks for the quick turnaround.

**Comment by JituRewar** [comment]
@vjkumar2756  thanks for reworking around them -> 
1.Fixed TS2300 Duplicate Type Blocker: 
2.Hardened Version Regex against Trailing Newlines:
3.Comprehensive Unit Tests for isReferencesValid:
4.Normalized Version Helper Text & Imports: 
5.Modal State & Validation Sync: 

that was asked by paoloantinori and now the code is looking correct , No concern from my side, LGTM 🚀.

**Comment by vjkumar2756** [comment]
@JituRewar thanks for review

**Comment by vjkumar2756** [comment]
@EricWittmann 
Hi, please check my PR. If everything looks good, please process it to the next steps. 
Thanks.

**Comment by vjkumar2756** [comment]
I solve this

**Comment by paoloantinori** [comment]
Thanks for the nudge. I re-checked the current head (816b149): the code side looks unchanged since the state we verified earlier — the validation rework and the review-feedback fixes from bc22d2d are all still in place, so nothing new to review there.

However the two blocking items from the triage report are still open:

1. **DCO sign-off** — commits `1305392` and `bc22d2d` still have no `Signed-off-by:` trailer. Running `git rebase --signoff main` (or amending those two commits with `-s`) and force-pushing will clear it. Commenting "I solve this" does not change the commits — the trailer has to be in the commit messages themselves.
2. **Linked issue sign-off** — #9595 still has no maintainer approval. That one is not yours to fix directly; it needs a maintainer to comment on the issue. Worth pinging one explicitly there so it does not block the triage.

Once the signed-off commits are pushed, `/retry` should turn the report green and a maintainer can pick it up.

**Comment by vjkumar2756** [comment]
Hi @EricWittmann,The triage bot is currently flagging Issue #9595 because it needs a maintainer comment on the issue itself to clear triage/red. Could you please add a quick comment on #9595 .

**Comment by vjkumar2756** [comment]
Hi  @EricWittmann  , could you please take a look at my PR when you get a chance. If everything looks good, I’d appreciate it if you could review it and guide me on the next steps. 
Thank you!

**Comment by vjkumar2756** [comment]
hi @carlesarnal 
could you please take a look at my PR when you get a chance. If everything looks good, I’d appreciate it if you could review it and guide me on the next steps.
Thank you!

**Comment by malladinagarjuna2** [comment]
hey @vjkumar2756  kindly rebase your pr, many irrelevant commits are coming.
